src/main/java/com/cloudbees/hudson/plugins/folder/computed/ThrottleComputationQueueTaskDispatcher.java

```diff
@@ -92,4 +134,14 @@ public int indexingCount(@CheckForNull Node node) {
         return result;
     }
 
+    private static class NonBlockedDetails {
```

Comment:
Maybe expressed more easily using a `WeakHashMap<Queue.Item, Long>`?

Reply:
Actually I think there's an even better way to express this